src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CspKeyContainerInfo.cs

```diff
@@ -51,16 +47,15 @@ public bool Accessible
         {
             get
             {
-                // This method will pop-up a UI for hardware keys.
-                SafeProvHandle safeProvHandle = SafeProvHandle.InvalidHandle;
-                int hr = CapiHelper.OpenCSP(_parameters, (uint)CapiHelper.CryptAcquireContextFlags.CRYPT_SILENT, ref safeProvHandle);
-                if (hr != CapiHelper.S_OK)
+                object retVal = ReadKeyParameterSilent(Constants.CLR_ACCESSIBLE, throwOnNotFound: false);
+
+                if (retVal == null)
                 {
+                    // The key wasn't found, so consider it to be not accessible.
```

Comment:
I'm not clear on whether this is the right answer -- if we have a provider, it must exist in some capacity. I wonder if this is about ACLs or something.